genie-core/src/main/java/com/netflix/genie/core/jobs/workflow/impl/InitialSetupTask.java

```diff
@@ -376,6 +382,60 @@ void createJobEnvironmentVariables(
         writer.write(LINE_SEPARATOR);
     }
 
+    @VisibleForTesting
+    void createJobRequestEnvironmentVariables(
+        final Writer writer,
+        final JobRequest jobRequest
+    ) throws IOException {
+
+        // create environment variable for the command tags/criteria in the job request
+        writer.write(JobConstants.EXPORT
+            + JobConstants.GENIE_REQUESTED_COMMAND_TAGS_ENV_VAR
+            + JobConstants.EQUALS_SYMBOL
+            + JobConstants.DOUBLE_QUOTE_SYMBOL
+            + tagsToString(jobRequest.getCommandCriteria())
+            + JobConstants.DOUBLE_QUOTE_SYMBOL
+            + LINE_SEPARATOR);
+
+        final List<ClusterCriteria> clusterCriterias = jobRequest.getClusterCriterias();
+
+        final List<String> clusterCriteriasStrings = new ArrayList<>(clusterCriterias.size());
+
+        for (ClusterCriteria clusterCriteria : clusterCriterias) {
+            clusterCriteriasStrings.add("[" + tagsToString(clusterCriteria.getTags()) + "]");
+        }
+
+        // Append new line
+        writer.write(System.lineSeparator());
+
+        // create environment variable for the list of cluster tags/criteria in the job request as a single
+        // value in the form "[[x,y,z],[a,b,c]]"
+        writer.write(JobConstants.EXPORT
```

Comment:
would it make more sense to use a format that's already known like JSON?

Additionally perhaps for scale (and I think we have a JIRA for this already) a better feature would be instead of continually dumping environment variables we just write out JSON dumps of the various components (job request, cluster, command, applications) to disk as files and then users can do whatever they want with them. Then we don't have to change all the time or worry about this formatting. Cause there is every chance this could eventually have some bash issues with quotes etc.

Reply:
This is JSON-ish (it's missing the quotes around strings because we'd need to escape them, making it harder to use this variable from bash.

Are you suggesting removing these 2 variables and instead dumping the entire job request as JSON, in this patch?

Reply:
I'm wondering out loud if rather than doing this and then having @rdblue and others write scripts to this specification and then they decide they want something else and we go to another if it wouldn't just be worth it to short circuit all this and give as much info as possible up front. Not saying we have to do it but to me that seems to be the most scalable / flexible.

Reply:
Plus if we dump all the info into files as JSON every possible language would know how to parse it whereas this format needs to be parsed specifically

Reply:
Right now, I don't consider this a specification. My code to match is:

```
if echo $COMMAND_TAGS | grep 'data:test'; then
  # test metastore
else
  # prod metastore
fi
```

Whatever you guys want to define as the convention is fine with me. I think the original request was to just dump the request JSON to a file. That would work.